File: src/model/Statements/CompoundStatement.java
package model.Statements;

import interfaces.IStatement;

/**
 * Created by Lucian on 10/11/2015.
 */
public class CompoundStatement implements IStatement {

    /**
     * First statement
     */
    private IStatement firstStatement;

    /**
     * Second statement
     */
    private IStatement secondStatement;

    /**
     * The constructor
     * @param firstStatement The first statement
     * @param secondStatement The second statement
     */
    public CompoundStatement(IStatement firstStatement, IStatement secondStatement) {
        this.firstStatement = firstStatement;
        this.secondStatement = secondStatement;
    }

    /**
     * Getters and setters
     */
    public IStatement getFirstStatement() {
        return firstStatement;
    }

    public void setFirstStatement(IStatement firstStatement) {
        this.firstStatement = firstStatement;
    }

    public IStatement getSecondStatement() {
        return secondStatement;
    }

    public void setSecondStatement(IStatement secondStatement) {
        this.secondStatement = secondStatement;
    }

    /**
     * String representation
     */
    @Override
    public String toString() {
        return firstStatement.toString() + ";" + secondStatement.toString();
    }
}

File: C#/ToyLanguage/ToyLanguage/Model/Statements/CompoundStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguage.Interfaces;

namespace ToyLanguage.Model.Statements
{
    [Serializable]
    class CompoundStatement:IMyStatement
    {
        /**
    * First statement
    */
        private IMyStatement firstStatement;

        /**
         * Second statement
         */
        private IMyStatement secondStatement;

        /**
         * The constructor
         * @param firstStatement The first statement
         * @param secondStatement The second statement
         */
        public CompoundStatement(IMyStatement firstStatement, IMyStatement secondStatement)
        {
            this.firstStatement = firstStatement;
            this.secondStatement = secondStatement;
        }

        /**
         * Getters and setters
         */
        public IMyStatement getFirstStatement()
        {
            return firstStatement;
        }

        public void setFirstStatement(IMyStatement firstStatement)
        {
            this.firstStatement = firstStatement;
        }

        public IMyStatement getSecondStatement()
        {
            return secondStatement;
        }

        public void setSecondStatement(IMyStatement secondStatement)
        {
            this.secondStatement = secondStatement;
        }

        /**
         * String representation
         */
 
    public string MyToString()
        {
            return firstStatement.MyToString() + ";" + secondStatement.MyToString();
        }

        public ProgramState execute(ProgramState programState)
        {
            programState.getExecutionStack().push(secondStatement);
            programState.getExecutionStack().push(firstStatement);
            return null;
        }
    }
}
